packages/m2m-gateway/test/integration/purposes/getPurposes.test.ts

```diff
@@ -0,0 +1,80 @@
+import { describe, it, expect, vi, beforeEach } from "vitest";
+import { m2mGatewayApi, purposeApi } from "pagopa-interop-api-clients";
+import { genericLogger } from "pagopa-interop-commons";
+import {
+  expectApiClientGetToHaveBeenCalledWith,
+  mockInteropBeClients,
+  purposeService,
+} from "../../integrationUtils.js";
+import { PagoPAInteropBeClients } from "../../../src/clients/clientsProvider.js";
+import {
+  getMockM2MAdminAppContext,
+  getMockedApiPurpose,
+} from "../../mockUtils.js";
+import {
+  toGetPurposesApiQueryParams,
+  toM2MGatewayApiPurpose,
+} from "../../../src/api/purposeApiConverter.js";
+import { WithMaybeMetadata } from "../../../src/clients/zodiosWithMetadataPatch.js";
+
+describe("getPurposes", () => {
+  const mockParams: m2mGatewayApi.GetPurposesQueryParams = {
+    eserviceIds: [],
+    offset: 0,
+    limit: 10,
+  };
+
+  const mockApiPurpose1 = getMockedApiPurpose();
+  const mockApiPurpose2 = getMockedApiPurpose();
+
+  const mockApiPurposes = [mockApiPurpose1.data, mockApiPurpose2.data];
+
+  const mockPurposeProcessResponse: WithMaybeMetadata<purposeApi.Purposes> = {
+    data: {
+      results: mockApiPurposes,
+      totalCount: mockApiPurposes.length,
+    },
+    metadata: undefined,
+  };
+
+  const mockGetPurposes = vi.fn().mockResolvedValue(mockPurposeProcessResponse);
+
+  mockInteropBeClients.purposeProcessClient = {
+    getPurposes: mockGetPurposes,
+  } as unknown as PagoPAInteropBeClients["purposeProcessClient"];
+
+  beforeEach(() => {
+    mockGetPurposes.mockClear();
+  });
+
+  it("Should succeed and perform API clients calls", async () => {
```

Comment:
Add also a test case where you make the function `throw` the `purposeNotFound` error you added